ocp_resources/resource.py

```diff
@@ -1028,6 +1033,58 @@ def wait_for_condition(self, condition: str, status: str, timeout: int = 300) ->
                     if cond["type"] == condition and cond["status"] == status:
                         return
 
+    def wait_for_status_condition(
```

Comment:
I don't understand how this belongs in wrapper. It seems more specific to tests/infra that uses wrapper. e.g. in cnv-tests we have wait_for_consistent_resource_conditions(), if you need more control, may be consider creating a helper there?

Reply:
Good point, I also reviewed [wait_for_consistent_resource_conditions](https://github.com/RedHatQE/cnv-tests/blob/5611e099b05265688a2c93b922bbedfa3d834c07/utilities/infra.py#L411) and it doing most of the functionalities my PR introduces.
I can extend it there.